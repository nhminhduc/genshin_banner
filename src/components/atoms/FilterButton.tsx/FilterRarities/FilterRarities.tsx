import Star from "assets/images/Star";
import FilterButton from "components/atoms/FilterButton.tsx/FilterButton";
import { config } from "config";
import { useFilterContext } from "hooks/useFilterContext";
import { isEqual } from "lodash";

const FilterRarities = () => {
  const { rarityFilter, setRarityFilter } = useFilterContext();

  const rarities = config.rarities;

  const onRarityFilterButtonClick = (rarityValue: string, isChecked: boolean) => {
    if (isEqual(rarityFilter, rarities)) {
      return setRarityFilter([rarityValue]);
    }
    if (isChecked === true) {
      return setRarityFilter([rarityValue, ...rarityFilter]);
    }
    if (rarityFilter.length === 1) {
      return setRarityFilter(rarities);
    }
    return setRarityFilter(rarityFilter.filter((element) => element !== rarityValue));
  };

  return (
    <div>
      {rarities.map((rarity) => (
        <FilterButton
          id={rarity}
          key={rarity}
          label={rarity}
          name="rarityFilter"
          onButtonClick={onRarityFilterButtonClick}
          icon={<Star />}
          chosen={rarityFilter.includes(rarity)}
        />
      ))}
    </div>
  );
};

export default FilterRarities;
