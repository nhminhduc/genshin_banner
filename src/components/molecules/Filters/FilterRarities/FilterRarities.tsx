import Star from "assets/images/Star";
import FilterButton from "components/atoms/FilterButton/FilterButton";
import { config } from "config";
import { useFilterContext } from "hooks/useFilterContext";
import { isEqual } from "lodash";

const FilterRarities = () => {
  const { rarityFilter, setRarityFilter } = useFilterContext();

  const { rarities } = config;

  const onRarityFilterButtonClick = (
    rarityValue: string,
    isChecked: boolean,
  ) => {
    if (isEqual(rarityFilter, rarities)) {
      return setRarityFilter([rarityValue]);
    }
    if (isChecked === true) {
      return setRarityFilter([rarityValue, ...rarityFilter]);
    }
    if (rarityFilter.length === 1) {
      return setRarityFilter(rarities);
    }
    return setRarityFilter(
      rarityFilter.filter((element) => element !== rarityValue),
    );
  };

  return (
    <div className="m-2">
      {rarities.map((rarity) => (
        <FilterButton
          chosen={rarityFilter.includes(rarity)}
          className="bg-amber-700 border border-amber-200 rounded w-12 my-1 text-white font-bold"
          icon={<Star className="h-4 w-4 md:h-5 md:w-5" />}
          id={rarity}
          key={rarity}
          label={rarity}
          name="rarityFilter"
          onButtonClick={onRarityFilterButtonClick}
        />
      ))}
    </div>
  );
};

export default FilterRarities;
