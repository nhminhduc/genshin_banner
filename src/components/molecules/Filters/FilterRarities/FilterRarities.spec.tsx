import { render } from "@testing-library/react";
import CharacterContextProvider from "context/CharacterContext";
import FilterContextProvider from "context/FilterContext";
import { describe, expect, test } from "vitest";

import FilterRarities from "./FilterRarities";

describe("FilterRarities test", () => {
  test("FilterRarities load", () => {
    const { container } = render(
      <FilterContextProvider>
        <CharacterContextProvider>
          <FilterRarities />
        </CharacterContextProvider>
      </FilterContextProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
