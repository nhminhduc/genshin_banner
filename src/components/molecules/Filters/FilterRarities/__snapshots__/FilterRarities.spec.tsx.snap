// Vitest Snapshot v1

exports[`FilterRarities test > FilterRarities load 1`] = `
<div>
  <div
    class="m-2"
  >
    <button
      class="flex justify-around items-center bg-amber-700 border border-amber-200 rounded w-12 my-1 text-white font-bold"
      id="4"
      name="rarityFilter"
      type="button"
    >
      4
       
      <svg
        class="h-4 w-4 fill-[#f7d04d] lg:h-5 lg:w-5"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M381.2 150.3l143.7 21.2c11.9 1.7 21.9 10.1 25.7 21.6 3.8 11.6.7 24.2-7.9 32.8L438.5 328.1l24.6 146.6c2 12-2.9 24.2-12.9 31.3-9.9 7.1-23 8-33.7 2.3l-128.4-68.5-128.3 68.5c-10.8 5.7-23.9 4.8-33.8-2.3-9.9-7.1-14.9-19.3-12.8-31.3l24.6-146.6L33.58 225.9c-8.61-8.6-11.67-21.2-7.89-32.8 3.77-11.5 13.74-19.9 25.73-21.6L195 150.3l64.4-132.33C264.7 6.954 275.9-.04 288.1-.04c12.3 0 23.5 6.994 28.8 18.01l64.3 132.33z"
        />
      </svg>
    </button>
    <button
      class="flex justify-around items-center bg-amber-700 border border-amber-200 rounded w-12 my-1 text-white font-bold"
      id="5"
      name="rarityFilter"
      type="button"
    >
      5
       
      <svg
        class="h-4 w-4 fill-[#f7d04d] lg:h-5 lg:w-5"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M381.2 150.3l143.7 21.2c11.9 1.7 21.9 10.1 25.7 21.6 3.8 11.6.7 24.2-7.9 32.8L438.5 328.1l24.6 146.6c2 12-2.9 24.2-12.9 31.3-9.9 7.1-23 8-33.7 2.3l-128.4-68.5-128.3 68.5c-10.8 5.7-23.9 4.8-33.8-2.3-9.9-7.1-14.9-19.3-12.8-31.3l24.6-146.6L33.58 225.9c-8.61-8.6-11.67-21.2-7.89-32.8 3.77-11.5 13.74-19.9 25.73-21.6L195 150.3l64.4-132.33C264.7 6.954 275.9-.04 288.1-.04c12.3 0 23.5 6.994 28.8 18.01l64.3 132.33z"
        />
      </svg>
    </button>
  </div>
</div>
`;
